#Cutadapt to remove primer sequences from Nanopore fastq
cutadapt -g Fwd1_1=^TTCTGGTGATGCTTGTGCTC -g Fwd1_2=^AAAAGGACAAGCAGCGAAAA -g Fwd1_3=^ATGCCTACATTCTATCTTGCGTTAC -g Fwd1_4=^CTAGAGGTCCGCGAGATTTG -g Fwd1_5=^CGACATTGACAGCCTTCTCA -g Fwd1_6=^TTGCTCCATCTGTCAGCAAC -g Fwd1_7=^GGACATCTCTGGCTCGAAAG -g Fwd1_8=^TCCAGGCTGTTGGAGAACACTTCA -g Fwd1_9=^CAGACGGTGGCGTATATGAG -g Fwd1_10=^GCGAGCCATAAAGCAGTTTC -g Fwd1_11=^GCCTTCTTTGACCAGCTTTG -g Fwd1_12=^AGTGCGGAGGAACTGCTAAA -g Fwd1_13=^TCCAAGGTGACCCCTGTTAG -g Fwd1_14=^CCCATGTTGTCACGTCACTC -g Fwd1_15=^TACGGGGCACTTAACCTGAC -g Fwd1_16=^GGCACCATAGCATGTCACAC -g Fwd1_17=^CCCGTTCACCAAAACAGTCT -g Fwd1_18=^ACCTCCCATAGCAACACCAG -g Fwd1_19=^CCAGACATACCCCAAACCAC -g Fwd1_20=^GCCCGTTGGGTTACATTAAGGTGT -g Fwd1_21=^CTTTGGGTTCCATTGTGTGCCCTT -g Fwd1_22=^ACGCCATACCCAAGTGAGTC -g Fwd1_24=^ACGCCGAGTCATCTCTCATTTGGA -g Fwd1_25=^GTGCAGAGCCTTGACATTGA -g EBNA2_Fw_Extra=^TGGGAATGGTGTTAACTTTC -g Fwd2_1=^CTGTTTATGAGACGCCAGCA -g Fwd2_2=^TTATGGTTCAGTGCGTCGAG -g Fwd2_3=^AGGGATGCCTGGACACAAGA -g Fwd2_4=^GCAGGCAGTACGAGATGTCA -g Fwd2_5=^TGCTCCTGATGTTTCTGAGGTGGA -g Fwd2_6=^GGTGACCACTGAGGGAGTGT -g Fwd2_7=^TCAGGAGGTCGTCAAAATCC -g Fwd2_8=^CCAGTCGCCGTTACTCATCT -g Fwd2_9=^GCCTCTATGTCGCTCTGACC -g Fwd2_10=^CTCGCGTGTTAGGAAGGAAG -g Fwd2_11_t2=^ATACATAGGAGCCTCACGAA -g Fwd2_11=^GGTGAAACGCGAGAAGAAAG -g Fwd2_12=^CCCACCACGTCTTCAACTTT -g Fwd2_12_2=^ACTCCCGGCTGTAAATTCCT -g Fwd2_13=^ACAGACCATCTACGCCAACC -g Fwd2_14=^GATGTTGCTGGGGCTAATGT -g Fwd2_15=^CGTTGGAAGTTGTTGGGACT -g Fwd2_16=^GGTCTCAACGTGTCCTGGTT -g Fwd2_17=^CCTGAGAACGCTCCAGGTAG -g Fwd2_18=^TTTGGGATGCATCACTTTGA -g Fwd2_19=^TCGTGGCTCGTACAGACGATTGTT -g Fwd2_20=^CCCACACCTTCACTCCTTGT -g Fwd2_21=^TGGAAGAAGGCGTAGAGCAT -g Fwd2_22=^AGGTTGCACACCACATCAAA -g Fwd2_23=^CACGGGGTTTATGTTTCTGG -g Fwd_sWGA1=^GCCGCCG -g Fwd_sWGA2=^GCGGCGG -g Fwd_sWGA3=^CGAGACC -g Fwd_sWGA4=^GGCCCGC -g Fwd_sWGA5=^GGTGGCG -g Fwd_sWGA6=^GTGGCGG -g Fwd_sWGA7=^CGGCCAC -g Fwd_sWGA8=^ACGGCCC -g Fwd_sWGA9=^GACCCCG -g Fwd_sWGA10=^CCCGGAC -a Rev1_1=TGCTGGCGTCTCATAAACAG$ -a Rev1_2=GTGCAGGAGGCTGTTTCTTC$ -a Rev1_3=TTACTGGATGGAGGGGCGAGGTCTT$ -a Rev1_4=AGAAGGCAAGCGAAAATTGA$ -a Rev1_5=AAACACGAATGCCAAGAACC$ -a Rev1_6=CACAAGCCTCCTCTCAGGAC$ -a Rev1_7=AGGAGGAGAACCCGAGGATA$ -a Rev1_7_t2=AGGAGGAGAACCCGAGGATC$ -a Rev1_8=ATCACAGTCACCCCCAGAAG$ -a Rev1_9=CAAAGAGCCCCGTAAAGATG$ -a Rev1_10=TCTCCCGAACTAGCAGCATT$ -a Rev1_11=GACGGGTTCTACTGGCATGT$ -a Rev1_11_t2=GACGGGTTCTACTGGCATGG$ -a Rev1_12=TGCAGAGGATGAGACCAGTG$ -a Rev1_13=TGATGCAGAGTCGCCTAATG$ -a Rev1_14=CACCGTGTTGGAGACCTTTT$ -a Rev1_15=TGACGGAGCTGTATCACGAG$ -a Rev1_16=AGTCCCAACAACTTCCAACG$ -a Rev1_17=AACCAGGACACGTTGAGACC$ -a Rev1_18=CCCGTGCGATGAGTTTATTT$ -a Rev1_19=CTCCAGAGGGCAGACGTTAG$ -a Rev1_20=CATGCAGTGGTGTCAGACAGGAAA$ -a Rev1_21=TTTGCGCCTTCTCCTGGTTTATGC$ -a Rev1_23=TCAAGAACCTGACGGAGCTT$ -a Rev1_24=CGTGACTACCCCCACGTACT$ -a Rev1_25=TGAACACCACCACGATGACT$ -a EBNA2_Rev_Extra=ATGTGTTGTGTGTGGTTTTG$ -a Rev2_1=TTTTCGCTGCTTGTCCTTTT$ -a Rev2_2=GAACTGAGGAGGGCATGAAG$ -a Rev2_3=AACATGGACTGGGAGTGGAG$ -a Rev2_4=TCCCTTCACATCCCAGAGAC$ -a Rev2_5=AGGTAACTTCTTTGAGCCTCCCGA$ -a Rev2_6=ATTTCAGGACTACCTGCGCGACTT$ -a Rev2_7=TTTCACATCCGACTCATTCCCTGC$ -a Rev2_8=ACCTTTCATCCGAACTCCTCAGGT$ -a Rev2_9=CGGAGGCGTGGTTAAATAAA$ -a Rev2_10=AGGCAAAGCTGGTCAAAGAA$ -a Rev2_11=TTTAGCAGTTCCTCCGCACT$ -a Rev2_12=CCATACCAGGTGCCTTTTGT$ -a Rev2_12_2=TGGCCAGAAATACACCAACA$ -a Rev2_13=CCACCACAAGAAGGTGTCCT$ -a Rev2_14=AGAGAGGGAGTTTCGCTTCC$ -a Rev2_15=CATTTTACCAGGGACGAGGA$ -a Rev2_16=GTGAAGGTATGTGCCGGTCT$ -a Rev2_17=CCTGGTGAGAAGTTGGTGGT$ -a Rev2_18=CCTCAAAGGTGTGGTCGTTT$ -a Rev2_19=ACCTGGTACATTGTGCCCATCAGA$ -a Rev2_20=CAGAGCCAGGCACATCTACA$ -a Rev2_21=GCAAGGCTGACTCACCTGTTTGA$ -a Rev2_22=GACTCGCTCACCCAAGAAAG$ -a Rev2_23=CCCCCTCCACTTTTTCCA$ -a Rev_sWGA1=GCCGCCG$ -a Rev_sWGA2=GCGGCGG$ -a Rev_sWGA3=CGAGACC$ -a Rev_sWGA4=GGCCCGC$ -a Rev_sWGA5=GGTGGCG$ -a Rev_sWGA6=GTGGCGG$ -a Rev_sWGA7=CGGCCAC$ -a Rev_sWGA8=ACGGCCC$ -a Rev_sWGA9=GACCCCG$ -a Rev_sWGA10=CCCGGAC$ --untrimmed-output sample.nopcrprimers.fastq.gz -o sample.pcrprimersremoved.fastq.gz nanopore_input.fastq.gz
cutadapt -u 10 -o sample.nopcrprimers.first10basestrimmed.fastq.gz sample.nopcrprimers.fastq.gz
cutadapt -u -10 -o sample.nopcrprimers.firstlast10basestrimmed.fastq.gz sample.nopcrprimers.first10basestrimmed.fastq.gz  
cat sample.pcrprimersremoved.fastq.gz sample.nopcrprimers.firstlast10basestrimmed.fastq.gz > sample.cutadapt.fastq.gz

#Flye command
python bin/flye --nano-raw sample.cutadapt.fastq.gz --meta --out-dir output_directory --threads 120

#Canu command (whole genome assembly)
./canu gridOptions=" --partition=batch --ntasks=1 --time=80:00:00 --mem-per-cpu=40g --cpus-per-task=2 " -p output_directory_name maxInputCoverage=1000000000000 corMhapSensitivity=high corMinCoverage=0 redMemory=32 oeaMemory=32 batMemory=200 genomeSize=170k corOutCoverage=1000000000000 minReadLength=200 minOverlapLength=100 -nanopore sample.cutadapt.fastq.gz

#Canu command (Example for EBNA2 only for EBV Type 1)
minimap2 -ax map-ont EBV1_NC_007605.fa sample.cutadapt.fastq.gz | samtools view -S -b | samtools sort > sample.cutadapt.sorted.bam
samtools index sample.cutadapt.sorted.bam
samtools view -h -b sample.cutadapt.sorted.bam NC_007605:35000-39000 | samtools sort >  sample.cutadapt.EBNA2only.sorted.bam
samtools fastq sample.cutadapt.EBNA2only.sorted.bam -F 0 > sample.rawreads.cutadapt.EBNA2only.sorted.fastq
./canu gridOptions=" --partition=batch --ntasks=1 --time=2:00:00 --mem-per-cpu=20g --cpus-per-task=2 " -p output_directory_name maxInputCoverage=9999999 corOutCoverage=9999999 corMhapSensitivity=high corMinCoverage=0  genomeSize=1472  minReadLength=40 minOverlapLength=40 -nanopore sample.rawreads.cutadapt.EBNA2only.sorted.fastq

#Merging assemblies (Example for EBV Type 1)

#Combine all assemblies (Flye, Canu) into one fastq and map to EBV Type 1 reference genome as bam file called sample_contigs.sorted.bam
#Map all raw reads to EBV Type 1 Reference Genome as bam file called sample_rawreads.sorted.bam
pip install pybedtools==0.6.9
pip install pysam==0.8.1
module load biopython/1.73
module load numpy/intel_1.15.1
python2 jbam_tools_v03.py ref_merge_contigs -n NC_007605 -t 12 --regions_to_mask NC_007605_repeatMask.bed -r sample_rawreads.sorted.bam -c sample_contigs.sorted.bam -u 10 -f EBV1_NC_007605.fa -g 1 --min_unique_bp 60 -o ref_merge_contigs_sample
